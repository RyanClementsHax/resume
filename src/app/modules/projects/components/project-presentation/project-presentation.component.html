<article class="container">
  <img [src]="project.imgUrl" />
  <div class="details">
    <header class="details-header">
      <h2>{{ project.title }}</h2>
      <a *ngIf="project.projectUrl" [href]="project.projectUrl" class="url project-url" rel="noopener noreferrer"
        target="_blank">
        <i class="fas fa-external-link-alt"></i>
      </a>
    </header>
    <p>{{ project.description }}</p>
    <footer class="details-footer">
      <div class="technologies">
        <span class="tech" *ngFor="let tech of project.technologies">
          {{ tech }}
        </span>
      </div>
      <a *ngIf="project.repoUrl" [href]="project.repoUrl" class="url repo-url" rel="noopener noreferrer"
        target="_blank">
        <i class="fab fa-2x fa-github"></i>
      </a>
    </footer>
  </div>
</article>
