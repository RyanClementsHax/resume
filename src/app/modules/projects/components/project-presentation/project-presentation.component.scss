@import "../../../../../variables.scss";

.container {
    display: flex;
    flex-direction: column;
    align-items: center;
    max-width: $content-width;
    padding: $content-padding;
    border: $light-border;
    border-radius: 3px;
    margin: 0 auto 50px;
    box-shadow: $default-box-shadow;

    img {
        width: 100%;
        max-width: 100%;
        border-radius: 3px;
    }

    .description {
        display: inline-block;
        width: 100%;

        h2 {
            margin: 0;
        }
    }

    @media screen and (min-width: $small-screen-break-point) {
        flex-direction: row;
        align-items: flex-start;

        img {
            width: 30%;
        }

        .description {
            margin-left: $content-padding;
        }
    }
}