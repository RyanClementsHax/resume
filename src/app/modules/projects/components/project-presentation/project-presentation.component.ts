import { Component, Input } from '@angular/core';
import { Project } from '../../services/projects.service';

@Component({
  selector: 'project-presentation',
  templateUrl: './project-presentation.component.html',
  styleUrls: ['./project-presentation.component.scss']
})
export class ProjectPresentationComponent {
  @Input() project: Project = {
    title: '',
    description: '',
    imgUrl: ''
  };
}
