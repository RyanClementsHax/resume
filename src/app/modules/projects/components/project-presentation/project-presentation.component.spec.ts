import { async, ComponentFixture, TestBed } from '@angular/core/testing';
import { NO_ERRORS_SCHEMA } from '@angular/core';

import { ProjectPresentationComponent } from './project-presentation.component';

describe('ProjectPresentationComponent', () => {
  let component: ProjectPresentationComponent;
  let fixture: ComponentFixture<ProjectPresentationComponent>;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      declarations: [ProjectPresentationComponent],
      schemas: [NO_ERRORS_SCHEMA]
    })
    .compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(ProjectPresentationComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });
});
