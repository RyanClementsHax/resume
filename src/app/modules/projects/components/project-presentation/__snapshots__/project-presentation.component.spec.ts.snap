// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectPresentationComponent should match snapshot 1`] = `
ProjectPresentationComponent {
  "project": Object {
    "description": "",
    "imgUrl": "",
    "technologies": Array [],
    "title": "",
  },
}
`;
